### PR TITLE
Pre-commit hook settings

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [x] Configuration Update
- [ ] Bump-up dependent library

## Description

- Pre-commit hook settings with formatters and linters
- Updated contributing guide accordingly

## Related Tickets & Documents

- Related Issue #[OLS-83](https://issues.redhat.com//browse/OLS-83)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] Unit tests passed locally.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Clone, change anything, any try to commit it

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,11 +33,12 @@ make install-deps
 # install dev/tests dependencies
 make install-deps-test
 
-# code formatting (run this as a pre-commit step for your code changes)
+# code formatting
+# (this is also run automaticall as part of pre-commit hook)
```
Comment (bparees) [inline CONTRIBUTING.md:37]:
```suggestion
# (this is also run automatically as part of the pre-commit hook)
```

by pre-commit hook do you mean the PR test job?  If so, call it PR pre-merge testing and not pre-commit hook.

if you do mean pre-commit hook, where is this setup?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -33,11 +33,12 @@ make install-deps
...
 make format
 
-# code style and docstring style (run this as a pre-commit step for your
-# code changes)
+# code style and docstring style
+# (this is also run automaticall as part of pre-commit hook)
```
Comment (bparees) [inline CONTRIBUTING.md:41]:
```suggestion
# (this is also run automatically as part of the pre-commit hook)
```

same question as above though

Comment (tisnik) [reply CONTRIBUTING.md:41]: ah my bad, the "dotfile" was ignored. sorry!!!

File: CONTRIBUTING.md (hunk 1 above)
Comment (tisnik) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (tisnik) [reply CONTRIBUTING.md]: fixed too @bparees , updated doc how to setup it

**Pull request merged**